packages/shell-api/src/aggregation-cursor.ts

```diff
@@ -69,19 +69,25 @@ export default class AggregationCursor extends ShellApiClass {
     return this._cursor.isClosed();
   }
 
-  isExhausted(): Promise<boolean> {
-    return this._cursor.isExhausted();
+  async isExhausted(): Promise<boolean> {
+    return this._cursor.isClosed() && !await this._cursor.hasNext();
   }
 
   @returnsPromise
-  itcount(): Promise<number> {
-    return this._cursor.itcount();
+  async itcount(): Promise<number> {
+    let count = 0;
+
+    while (await this.hasNext()) {
+      await this.next();
+      count++;
+    }
+
+    return count;
   }
 
   @returnType('AggregationCursor')
-  map(f): AggregationCursor {
-    this._cursor.map(f);
-    return this;
+  map(f: (doc: Document) => Document): AggregationCursor {
+    return new AggregationCursor(this._mongo, this._cursor.map(f));
```

Comment:
I’m less sure about these. The fact that these used to return the original cursor and that that was *tested* stumped me a bit, tbh. What would the use of a `.map()` method be that doesn’t actually let you use the mapped results? This was not intended, right? 

Fwiw, this also doesn’t match https://docs.mongodb.com/manual/reference/method/cursor.map/, which claims that this gives an array, but returning a new cursor feels like the most sensible thing to me.

Reply:
I think the original behaviour might have been taken from the driver: https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/cursor.js#L934

Your change makes lot of sense to me, but I think we may need some support from driver in order to make it work or?

Reply:
Ok, this got cleared up, `.map()` actually does something that does not match my intuition for that method name at all :)